hercules/simconfig.py

```diff
@@ -0,0 +1,459 @@
+                       
+    def _handle_noise(self):
+        
+        if self._noise_key in self._config_dict:
+
+            if (self._noise_floor_psd_key or self.__sNoiseTemp) in self._config_dict[self._noise_key]:
+                self._config_dict[self._generators_key].insert(-1, self._noise_key)
+
+            if (self._noise_floor_psd_key and self._noise_temperature_key) in self._config_dict[self._noise_key]:
+                #prefer noise temperature over noise psd
+                self._config_dict[self._noise_key].pop(self._noise_floor_psd_key)
+
+            if self._random_seed_key not in self._config_dict[self._noise_key]:
+                self._set(self._noise_key, self._random_seed_key, _get_rand_seed())
+                
+    def _adjust_paths(self):
+        
+        self._prefix(self._array_key, self._xml_filename_key, 
+                        str(OUTPUT_DIR_CONTAINER) + '/')
+        self._prefix(self._sim_key, self._egg_filename_key, 
+                        str(OUTPUT_DIR_CONTAINER) + '/')
+        self._prefix(self._array_key, self._tf_receiver_filename_key, 
+                        str(HEXBUG_DIR_CONTAINER)+'/Phase3/TransferFunctions/')
+    
+    # -------- public part --------
+    
+    @property
+    def config_dict(self):
+        return self._config_dict
+        
+    def set_xml(self, path):
+        
+        name = path.name
+        self._set(self._array_key, self._xml_filename_key, 
+                    str(OUTPUT_DIR_CONTAINER / name))
+                    
+    def make_config_file(self, output_path):
+        
+        with open(output_path, 'w') as outFile:
+            json.dump(self._config_dict, outFile, indent=2)
+        
+
+class SimConfig:
+    
+    def __init__(self,
+                sim_name,
+                kass_template = HEXBUG_DIR/'Phase3'/KASS_CONFIG_NAME,
+                seed_kass = None,
+                t_max = None,
+                x_min = None,
+                x_max = None,
+                y_min = None,
+                y_max = None,
+                z_min = None,
+                z_max = None,
+                theta_min = None,
+                theta_max = None,
+                geometry = None,
+                energy = None,
+                locust_template = HEXBUG_DIR/'Phase3'/LOCUST_CONFIG_NAME,
+                n_channels = None,
+                egg_filename = None,
+                record_size = None,
+                n_records = None,
+                v_range = None,
+                lo_frequency = None,
+                n_elements_per_strip = None,
+                n_subarrays = None,
+                zshift_array = None,
+                array_radius = None,
+                element_spacing = None,
+                tf_receiver_bin_width = None,
+                tf_receiver_filename = None,
+                xml_filename = None,
+                seed_locust = None,
+                noise_floor_psd = None,
+                noise_temperature = None):
```

Comment:
This is just a suggestion. Since this is a wrapper around the LocustConfig and KassConfig and other potential configurations, to avoid clustering of options, can we ask the user to change the parameters "on demand" instead of doing it during initialization? For example the user create a SimConfig class then use a "set" method to set the parameter, and that set method goes to specific program Config classes. All other parameters are set to default.

Reply:
I'm not sure what you exactly mean here. Do you mean that it should be possible to create a config via a "set" method that sets only a single parameter and all other parameters are default values?
If that's the point you can already do that. The only required parameter is the name, everything else will be set to the default values in the config files automatically. When you include additional parameters during initialization the default values will be overwritten.